### DISCUSSION TITLE
Use cURL to simulate XCPD Sender for Troubleshooting.

### DISCUSSION THREAD

**Discussion by rjcdoc** [nextgenhealthcare/connect#6133] [General]
I have an XCPD Sender channel that has been working for weeks and all of a sudden stopped working.  I get a generic java error message response: ERROR: Error invoking web service [WebServiceException: java.net.SocketException: Unexpected end of file from server]

The Web Service say my request are getting there, but length of 0 bytes.

I have tried the following to debug the issue.

- Restarting my Mirth Instance.
- Simple curl commands to test connectivity to the endpoint. (curl endpoint --verbose) response is curl: (1) Received HTTP/0.9 when not allowed
- Tried telnet web service port. Gave some data like a valid HTTP1.0 connection got response of PConnection closed by foreign host

Any other troubleshooting ideas to get to the root cause?

**Comment by rjkroll** [comment]
I found this thread that has a couple of possible causes and solutions, not sure if any apply here, but it mainly seems like this error is returned from the server you are sending data to closing the connection to early and not necessarily anything Mirth related:

[https://stackoverflow.com/questions/19824339/java-simple-code-java-net-socketexception-unexpected-end-of-file-from-server
](url)

**Reply by rjcdoc** [reply]
Thanks for that.  I did see that.  I did resolve this.  It was a protocol issue.  We were connecting via HTTP instead of HTTPS.  Making that change and applying the correct certificate, solved the problem.

**Comment by rjcdoc** [comment]
I have resolved this issue.
